### PR TITLE
New Minecraft Commands Cheat Sheet

### PR DESCRIPTION
**Pull request by 122131**
###### What does your Pull Request do (check all that apply)? 

Choose the most relevant items and use the following title template to name
your Pull Request.

- [x] New Instant Answer
    - [x] Cheat Sheets: **`New {Minecraft Commands} Cheat Sheet`**
    - [ ] Other: **`New {IA Name} Instant Answer`**
- [ ] Improvement
    - [ ] Bug fix: **`{IA Name}: Fix {Issue number or one-line description}`**
    - [ ] Enhancement: **`{IA Name}: {Description of Improvements}`**
- [ ] Non–Instant Answer
    - [ ] Other (Role, Template, Test, Documentation, etc.): **`{GoodieRole/Templates/Tests/Docs}: {Short Description}`**

###### Description of changes

Add a cheat sheet for Minecraft chat commands

###### People to notify (@mention interested parties)
@GuiltyDolphin @moollaza 

---

Instant Answer Page: https://duck.co/ia/view/minecraft_commands_cheat_sheet

[Maintainer](http://docs.duckduckhack.com/maintaining/guidelines.html): @122131

### CODE REVIEW

**Comment by daxtheduck** [comment]
Metadata placeholder post
            
*If you haven't already, please edit your first comment to include a link to the IA page at the bottom.*
            
---
            
**How to find your IA page**
- New Instant Answer
  -- follow the steps [here](https://duck.co/ia/new_ia) to create your IA page.
              
- Existing Instant Answer
  -- find your IA page from this [list](https://duck.co/ia)

**Comment by 122131** [comment]
The triggers can be improved, maybe using the "cheats" category editing the triggers.yaml file, but this cheat sheet contains also other non-cheats commands, so I don't know if this is a good idea...

**Comment by edgesince84** [comment]
thanks @122131 we'll take a look at this soon!

**Comment by daxtheduck** [comment]
Automated screenshot from [beta.duckduckgo.com](https://beta.duckduckgo.com/?q=Minecraft commands cheat sheet)
                
                [Mobile screenshot](https://images.duckduckgo.com/iu/?u=https://ia-screenshots.s3.amazonaws.com/minecraft_commands_cheat_sheet_mobile.png?m2bxozkNg5)

**Comment by tagawa** [comment]
Thanks a lot for creating this Instant Answer @122131. I've installed it on the beta server so now it's publicly available for testing before we merge: https://beta.duckduckgo.com/?q=Minecraft+commands+cheat+sheet&ia=cheatsheet&iax=1

**Comment by moollaza** [comment]
@122131 is there any better way to categorize the commands so that the sections aren't so large?

Otherwise Thanks! This LGTM :+1:

**Comment by 122131** [comment]
@moollaza I can categorize them by "commands that modify blocks/entities/players/world", but there are many commands that belong to multiple categories...

A table (like the one shown in the source link) may be better but, as I know, there is no way to use a table In a cheat sheet, right?

**Comment by GuiltyDolphin** [comment]
@122131 Ideally we'd be able to trigger this on `minecraft cheats` or `minecraft commands` - the existance of the `minecraft` cheat sheet currently prevents this, unfortunately, it would be nice to support multiple cheat sheets with the same 'name' (topic/subject) w/o clashing triggers.

@zachthompson Do you think this would be worth investigating? There's a couple of issues that spring to mind, such as 'which cheat sheet triggers on just "cheat sheet"', and 'how do you decide which is the *main* cheat sheet' which would need to be resolved.

**Comment by 122131** [comment]
@GuiltyDolphin I was thinking of making other cheat sheets about minecraft and other games. It's better to wait until the IAs will trigger correctly or can I start building them?

**Comment by GuiltyDolphin** [comment]
@122131 This is a general triggering issue that (if we fix it) would likely require us to write some scripts for updating all cheat sheets - so go ahead, just be weary of the triggering :wink:

File: share/goodie/cheat_sheets/json/minecraft-commands.json [hunk 1]
```diff
@@ -0,0 +1,270 @@
...
+                "key": "\/effect <player> <effect> \\[seconds\\] \\[amplifier\\] \\[hideParticles\\]",
+                "val": "Add or remove status effects"
+            },
+            {
+                "key": "\/enchant <player> <enchantment ID> \\[level\\]",
+                "val": "Enchant a player item"
+            },
+            {
+                "key": "\/entitydata <entity> <dataTag>",
+                "val": "Modifie the data tag of an entity"
```
Comment (tagawa) [inline share/goodie/cheat_sheets/json/minecraft-commands.json]: Small typo: "Modifie" should be "Modify"

File: share/goodie/cheat_sheets/json/minecraft-commands.json [hunk 2]
```diff
@@ -0,0 +1,270 @@
...
+                "key": "\/weather <clear|rain|thunder> \\[duration\\]",
+                "val": "Set the weather"
+            },
+            {
+                "key": "\/worldborder <...>",
+                "val": "Manage the world border"
+            },
+            {
+                "key": "\/xp <amount>\\[L\\] \\[player\\]",
+                "val": "Manage the world border"
```
Comment (tagawa) [inline share/goodie/cheat_sheets/json/minecraft-commands.json]: The previous key is also "Manage the world border" - is this correct?

File: share/goodie/cheat_sheets/json/minecraft-commands.json [hunk 3]
```diff
@@ -0,0 +1,270 @@
...
+                "key": "\/spreadplayers <x> <z> <spreadDistance> <maxRange> <respectTeams> <player …>",
+                "val": "Teleport entities to random locations"
+            },
+            {
+                "key": "\/stats <...>",
+                "val": "Update objectives from command results"
+            },
+            {
+                "key": "\/stopsound <player> \\[source\\] \\[sound\\]",
+                "val": "Stop a sound"
```
Comment (tagawa) [inline share/goodie/cheat_sheets/json/minecraft-commands.json]: Please could this key/value be moved up next to the "Play a sound" key/value?

File: share/goodie/cheat_sheets/json/minecraft-commands.json [hunk 4]
```diff
@@ -0,0 +1,270 @@
...
+            },
+            {
+                "key": "\/summon <EntityName> \\[x\\] \\[y\\] \\[z\\] \\[dataTag\\]",
+                "val": "Summon an entity"
+            },
+            {
+                "key": "\/teleport <target entity> <x> <y> <z> \\[<y-rot> <x-rot>\\]",
+                "val": "Teleport entities"
```
Comment (tagawa) [inline share/goodie/cheat_sheets/json/minecraft-commands.json]: Please could this key/value be moved up next to the "Teleport entities to random locations" key/value?

File: share/goodie/cheat_sheets/json/minecraft-commands.json [hunk 5]
```diff
@@ -0,0 +1,270 @@
...
+                "key": "\/title <player> <clear|reset|subtitle <title>|times <fadeIn> <stay> <fadeOut>|title <title>>",
+                "val": "Manage screen titles"
+            },
+            {
+                "key": "\/toggledownfall",
+                "val": "Toggle the weather"
+            },
+            {
+                "key": "\/tp \\[target player\\] <<destination player>|<x> <y> <z> \\[<yaw> <pitch>\\]>",
+                "val": "Teleport players"
```
Comment (tagawa) [inline share/goodie/cheat_sheets/json/minecraft-commands.json]: Please could this key/value be moved up next to the "Teleport entities" key/value?

**Comment by tagawa** [comment]
@122131 Regarding the sectioning that @moollaza suggested, how about separating these items into new sections?

Blocks:
* Modify the data tag of a block 
* Copy blocks from one place to another 
* Fill a region with a specific block 
* Change a block to another block
* Test whether a block is in a location
* Test whether the blocks in two regions match

Teleporting:
* Teleport entities to random locations
* Teleport entities
* Teleport players

**Comment by daxtheduck** [comment]
Automated screenshot from [beta.duckduckgo.com](https://beta.duckduckgo.com/?q=Minecraft commands cheat sheet)
                
                [Mobile screenshot](https://images.duckduckgo.com/iu/?u=https://ia-screenshots.s3.amazonaws.com/minecraft_commands_cheat_sheet_mobile.png?f3dMQ6A85L)

**Comment by daxtheduck** [comment]
Automated screenshot from [beta.duckduckgo.com](https://beta.duckduckgo.com/?q=Minecraft commands cheat sheet)
                
                [Mobile screenshot](https://images.duckduckgo.com/iu/?u=https://ia-screenshots.s3.amazonaws.com/minecraft_commands_cheat_sheet_mobile.png?xJSrnr7umi)

File: share/goodie/cheat_sheets/json/minecraft-commands.json [hunk 6]
```diff
@@ -0,0 +1,280 @@
...
+                "key": "\/execute <entity> <x> <y> <z> detect <x2> <y2> <z2> <block> <data> <command …>",
+                "val": "Execute another command"
+            },
+            {
+                "key": "\/help \\[page|command name\\]",
+                "val": "Provide help for commands"
+            },
+            {
+                "key": "\/say <message …>",
+                "val": "Display a message to multiple players"
```
Comment (tagawa) [inline share/goodie/cheat_sheets/json/minecraft-commands.json]: Please could this key/value be moved up next to "Display a private message to other players"

File: share/goodie/cheat_sheets/json/minecraft-commands.json [hunk 7]
```diff
@@ -0,0 +1,280 @@
...
+                "val": "Set the time before idle players are kicked"
+            },
+            {
+                "key": "\/spawnpoint \\[\\[player\\] \\[<x> <y> <z>\\]\\]",
+                "val": "Set the spawn point for a player"
+            },
+            {
+                "key": "\/stopsound <player> \\[source\\] \\[sound\\]",
+                "val": "Stop a sound"
+            },
```
Comment (tagawa) [inline share/goodie/cheat_sheets/json/minecraft-commands.json]: Please could this key/value be moved up next to "Play a sound"

**Comment by tagawa** [comment]
Thanks for the updates @122131 - looking good! Please could you try combining the "Other" and "Various" sections into one section? Then I think we should be done.

**Comment by daxtheduck** [comment]
Automated screenshot from [beta.duckduckgo.com](https://beta.duckduckgo.com/?q=Minecraft commands cheat sheet)
                
                [Mobile screenshot](https://images.duckduckgo.com/iu/?u=https://ia-screenshots.s3.amazonaws.com/minecraft_commands_cheat_sheet_mobile.png?rYqdFb602o)

**Comment by daxtheduck** [comment]
Automated screenshot from [beta.duckduckgo.com](https://beta.duckduckgo.com/?q=Minecraft commands cheat sheet)
                
                [Mobile screenshot](https://images.duckduckgo.com/iu/?u=https://ia-screenshots.s3.amazonaws.com/minecraft_commands_cheat_sheet_mobile.png?mTCD8wNlXh)

**Comment by tagawa** [comment]
Thanks @122131. One final thing - please could the "Stop a sound" key/value be moved up next to the "Play a sound" one?

**Comment by daxtheduck** [comment]
Automated screenshot from [beta.duckduckgo.com](https://beta.duckduckgo.com/?q=Minecraft commands cheat sheet)
                
                [Mobile screenshot](https://images.duckduckgo.com/iu/?u=https://ia-screenshots.s3.amazonaws.com/minecraft_commands_cheat_sheet_mobile.png?u966xyxBN6)

**Comment by tagawa** [comment]
Thanks again @122131. Looks good to me on the beta server: https://beta.duckduckgo.com/?q=Minecraft%20commands%20cheat%20sheet&ia=cheatsheet
Let's give it a couple of days for possible further feedback from the community, and then we should be able to merge.

**Comment by 122131** [comment]
Thanks, @tagawa ! 
And also thank you to @moollaza and  @GuiltyDolphin for the suggestions! :smile: 
If there are others issues, let me know! :+1:

**Comment by tagawa** [comment]
This looks good on the beta server and there's been no further feedback, so let's merge. Thanks for creating this and implementing all the comments @122131!

**Comment by daxtheduck** [comment]
@122131 this is now merged! It should be live on DuckDuckGo within one week (or during the next release) and we'll be sure to notify you when it happens.

After your Instant Answer goes live for the World to use, you will be able to see [traffic information](http://docs.duckduckhack.com/maintaining/tools.html#monitor-your-instant-answer-page) about your IA as well as receive feedback directly from users. 
                
Please refer to the [maintenance guide and docs](http://docs.duckduckhack.com/maintaining/guidelines.html) to keep up with potential changes to your Instant Answer.
                
Thank you again for your help!

**Pull request merged**